src/MicrosoftTeams.ts

```diff
@@ -463,6 +501,43 @@ namespace microsoftTeams
             ensureInitialized(frameContexts.authentication);
 
             sendMessage("authentication.authenticate.failure", reason);
+            window.close();
+        }
```

Comment:
Didn't we discover a race condition here where the message wasn't actually sent since it destroyed itself?